code/game/machinery/firealarm.dm

```diff
@@ -144,8 +172,11 @@
 		return ..()
 	add_fingerprint(user)
 	play_click_sound("button")
-	var/area/A = get_area(src)
-	if(A.fire || A.party)
+	if(myarea.fire || myarea.party)
+		if(areafire_check() && !(obj_flags & EMAGGED))
+			to_chat(user, span_danger("There is a fire in this area, you cannot lift the fire doors."))
```

Comment:
Keep the ability to turn it off, this shit is annoying

Reply:
It's not

Reply:
fine